### ISSUE TITLE
access of inactive union field on aarch64

### ISSUE THREAD

**Issue by daurnimator** [ziglang/zig#4492]
Cut down repro of the error found by CI of #4304.

```zig
const std = @import("std");
const testing = std.testing;

pub const Token = union(enum) {
    Number,
    True,
};

fn parseInternal(comptime T: type, token: Token) !T {
    switch (@typeInfo(T)) {
        .Bool => {
            return switch (token) {
                .True => true,
                else => error.UnexpectedToken,
            };
        },
        .Float, .ComptimeFloat => {
            const numberToken = switch (token) {
                .Number => |n| n,
                else => return error.UnexpectedToken,
            };
            unreachable;
        },
        .Union => |unionInfo| {
            if (unionInfo.tag_type) |_| {
                // try each of the union fields until we find one that matches
                inline for (unionInfo.fields) |u_field| {
                    if (parseInternal(u_field.field_type, token)) |value| {
                        return @unionInit(T, u_field.name, value);
                    } else |err| {
                        // Bubble up error.OutOfMemory
                        // Parsing some types won't have OutOfMemory in their
                        // error-sets, for the condition to be valid, merge it in.
                        if (@as(@TypeOf(err) || error{OutOfMemory}, err) == error.OutOfMemory) return err;
                        // otherwise continue through the `inline for`
                    }
                }
                return error.NoUnionMembersMatched;
            } else {
                unreachable;
            }
        },
        else => unreachable,
    }
    unreachable;
}

test "tagged union" {
    const T = union(enum) {
        float1: f64,
        float2: f64,
        bool1: bool,
    };
    const r = try parseInternal(T, @as(Token, .True));
    testing.expectEqual(@TagType(T).bool1, @as(@TagType(T), r));
    testing.expectEqual(true, r.bool1);
}
```

```
$ zig test   -target aarch64v8a-linux --test-cmd qemu-aarch64 --test-cmd-bin tagged_union.zig
Test [1/1] test "tagged union"...access of inactive union field
/home/daurnimator/src/zig/tagged_union.zig:56:32: 0x21069b in test "tagged union" (test)
    testing.expectEqual(true, r.bool1);
                               ^
/home/daurnimator/src/zig/lib/std/special/test_runner.zig:47:28: 0x2322db in std.special.main (test)
        } else test_fn.func();
                           ^
/home/daurnimator/src/zig/lib/std/start.zig:250:37: 0x21123b in std.start.posixCallMainAndExit (test)
            const result = root.main() catch |err| {
                                    ^
/home/daurnimator/src/zig/lib/std/start.zig:114:5: 0x2110b3 in std.start._start (test)
    @call(.{ .modifier = .never_inline }, posixCallMainAndExit, .{});
    ^
qemu: uncaught target signal 6 (Aborted) - core dumped

Tests failed. Use the following command to reproduce the failure:
/home/daurnimator/src/zig/zig-cache/o/2zrHy-77N57j79zhikZBIXXqMXffR4V19eJ7SSIYck4pWHthod4d0MAV4Xjn2BLh/test
```

**Comment by LemonBoy** [comment]
I've explained what the root cause is in #4304 heh

**tl;dr**: LLVM9 bug, LLVM10 doesn't have this problem

**Comment by andrewrk** [comment]
Fixed with the merge of llvm10 branch: 93c7fa105f9cc4dbb36192469ecff47102d955d6
